### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by AyLim29**
This pull request contains my implementation of Task 7 (Create a classifier based on Siamese network to classify Alzheimer’s Disease of the ADNI brain dataset)

The accuracy on the test set is 74.91%.

This pull request contains the following:

```
modules.py: The source code of the components of the Siamese network. 
dataset.py: Contains the data loader for loading and preprocessing the ADNI dataset
train.py: Contains the source code for training, validating, testing and saving the model. Also contains code for plotting training performance.
predict.py: Example use of the trained model. It samples a random image from the test set and classifies it (and visualizes it). 
README.md: Project documentation (and /images: Stores images used in README.md)
```

Dependencies are detailed in README.md

Name: Aaron Lim

Student Number: 46420763

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Good
- Clear flow of information
- Algortihm and usage explained well
- Sufficient results/plots provided

Commit messages: OK

Code: 
- good design
- minimal comments
- classification is implemented as a prediction against a query image (similarity detection only), no separate classifier has been trained

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification of individual images without a query image**
- triplet network achieves ~75% for validation set
- model performance on the validation set seem very unstable, but no attempts have been made to resolve this

General comments:
- **algorithm solves the respective problem somewhat half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained triplet network

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem partially, no single image classifier -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, usage missing -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
